exe/facter.cc

```diff
@@ -299,6 +281,9 @@ int main(int argc, char **argv)
             log(level::warning, "Facter was called recursively, skipping external facts. Add '--no-external-facts' to silence this warning");
           } else {
             environment::set("INSIDE_FACTER", "true");
+            if (vm.count("external-dir")) {
+                external_directories.push_back(vm["external-dir"].as<string>());
```

Comment:
Edit: figured out how multiple directories could be specified on the command line. You have to use the flag multiples times, and each time it will concatenate into the vector. I assume we do not want to break this behavior. Luckily, this also seems to just work with hocon list syntax (yay @branan  being thorough with the program_options extension!), so I will go back to using `vector<string>`.